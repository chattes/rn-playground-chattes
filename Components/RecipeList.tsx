import React, {useEffect} from 'react';
import {StyleSheet, View, FlatList, ActivityIndicator} from 'react-native';
import {fetchRecipes} from '../Redux/actions/recipeActions';
import {useSelector, useDispatch} from 'react-redux';

const RecipeList = (): JSX.Element => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRecipes());
  }, [dispatch]);

  const {
    recipes: {loading, collection},
  } = useSelector((state) => state);
  console.log(collection);
  if (loading) {
    return (
      <View testID="recipe-list">
        <ActivityIndicator />
      </View>
    );
  }
  return <View testID="recipe-list"></View>;
};

const styles = StyleSheet.create({});

export default RecipeList;
