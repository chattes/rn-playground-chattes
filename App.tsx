import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  StatusBar,
  View,
  Text,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';
import configureStore from './Redux/store';
import {Provider} from 'react-redux';
import RecipeList from './Components/RecipeList';

const store = configureStore();

const App = (): JSX.Element => {
  return (
    <>
      <Provider store={store}>
        <StatusBar barStyle="dark-content" />
        <SafeAreaView>
          <ScrollView
            contentInsetAdjustmentBehavior="automatic"
            style={styles.scrollView}>
            <View style={styles.titleContainer}>
              <Text style={styles.title} testID="title">
                Recipies
              </Text>
              <RecipeList />
            </View>
          </ScrollView>
        </SafeAreaView>
      </Provider>
    </>
  );
};

const styles = StyleSheet.create({
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 30,
  },
  scrollView: {
    backgroundColor: Colors.lighter,
  },

  body: {
    backgroundColor: Colors.white,
  },
});

export default App;
