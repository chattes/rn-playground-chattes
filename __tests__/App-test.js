/**
 * @format
 */
import React from 'react';
import App from '../App';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import RecipeList from '../Components/RecipeList';

describe('Testing test Component', () => {
  it('1+1 should be 2', () => {
    expect(1 + 1).toBe(2);
  });
});

describe('<App />', () => {
  it('should render app', () => {
    const wrapper = shallow(<App />);
    expect(
      wrapper.findWhere((node) => node.prop('testID') === 'title'),
    ).toExist();
  });

  it('matches snapshot', () => {
    const tree = shallow(<App />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});

describe('<RecipeList />', () => {
  it('should match snapshot', () => {
    const tree = shallow(<RecipeList />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
  it('should return a list of recipes', () => {
    const wrapper = shallow(<RecipeList />);
    expect(
      wrapper.findWhere((node) => node.prop('testID') === 'recipe-list'),
    ).toExist();
  });
});
