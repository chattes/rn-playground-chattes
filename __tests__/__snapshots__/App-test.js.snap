// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<Fragment>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <StatusBar
      animated={false}
      barStyle="dark-content"
      showHideTransition="fade"
    />
    <ForwardRef(SafeAreaView)>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
              }
            }
            testID="title"
          >
            Recipies
          </Text>
        </View>
      </ScrollView>
    </ForwardRef(SafeAreaView)>
  </Provider>
</Fragment>
`;

exports[`<RecipeList /> should match snapshot 1`] = `
<View
  testID="recipe-list"
/>
`;
